src/rust/src/transaction_ffi.rs

```diff
@@ -99,18 +229,23 @@ pub extern "C" fn zcash_transaction_precomputed_free(precomputed_tx: *mut Precom
     }
 }
 
-/// Calculates a signature digest for the given transaction.
+/// This MUST match `NOT_AN_INPUT` in `src/script/interpreter.h`.
+const NOT_AN_INPUT: usize = 0xffffffff;
+
+/// Calculates a ZIP 244 signature digest for the given transaction.
+///
+/// `index` must be an index into the transaction's `vin`, or `NOT_AN_INPUT` for
+/// calculating the signature digest for shielded signatures.
+///
+/// `sighash_ret` must point to a 32-byte array.
 ///
 /// Returns `false` if any of the parameters are invalid; in this case, `sighash_ret`
 /// will be unaltered.
 #[no_mangle]
-pub extern "C" fn zcash_transaction_transparent_signature_digest(
+pub extern "C" fn zcash_transaction_zip244_signature_digest(
     precomputed_tx: *const PrecomputedTxParts,
     hash_type: u32,
     index: size_t,
-    script_code: *const c_uchar,
-    script_code_len: size_t,
-    value: i64,
     sighash_ret: *mut [u8; 32],
 ) -> bool {
```

Comment:
FFI functions should generally be thin wrappers just encapsulating the unsafe interface to C; there's a lot of application logic here that should probably be factored out to another safe function.

Reply:
This method is already about as thin as it can be. All this method is doing is taking the unsafe C interface, and converting the arguments to safe parameters for `v5_signature_hash`.